### PR TITLE
rspec enhancements

### PR DESCRIPTION
**Pull request by beanieboi**
as proposed in https://github.com/troessner/reek/issues/497 this is the PR for the enhancements.

i intentionally used 2 commits here because these are 2 different features to me. it also makes sense to me, when you look at reverting commits, if you want to revert the monkeypatching commit you don't want to necessarily revert the partial double check.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -52,4 +52,13 @@ def ast(*args)
   config.run_all_when_everything_filtered = true
   config.include FactoryGirl::Syntax::Methods
   config.include Helpers
+
+  config.disable_monkey_patching!
+
+  config.mock_with :rspec do |mocks|
+    # Prevents you from mocking or stubbing a method that does not exist on
```
Comment (andyw8) [inline spec/spec_helper.rb]: I think the comment isn't needed – the reader can refer to the RSpec documentation if they're not familiar with this method.

**Comment by beanieboi** [comment]
removed the comment from `spec_helper.rb`

**Comment by mvz** [comment]
Looks good to me :shipit:

**Comment by chastell** [comment]
:+1:

**Pull request merged**